DOC-14371 2.2 EA features in 1.1 SDK (#6)
couchbase/docs-analytics-sdk-dotnet: modules/hello-world/pages/start-using-sdk.adoc

[quickstart]
== Connecting and Executing a Query

The 1.1 {name-sdk} adds support for JWT and client certificate authentication, as well as a new "async" poll-based API that uses request handles to fetch results.
Introduced in the 2.2 release of self-managed Enterprise Analytics Server, this API eliminates the need for long-running server connections.

The examples in this first section of the page are for the standard API -- async on the client side -- working with all 2.x releases of Enterprise Analytics (with Server Async examples following in the <<#server-asynchronous-api,Server Async section>>).

> Note, you will still be able to use this API with 2.2+ releases of Enterprise Analytics, in addition to the new API.

Reviewer: It would be good to make the language around version numbers consistent throughout. There's nothing specifically in the Couchbase style guide about version numbers, but all the examples given put the version number after the product name.

I've gone with Server Async instead of "async" here, but use the marketing name for the Server Async feature, when decided.

Relevant style rules: [Use earlier/later to describe software versions](https://docs.couchbase.com/styleguide/word-list.html#e), [Write in the Present Tense](https://docs.couchbase.com/styleguide/general.html#write-in-the-present-tense), [Do not use "note that"](https://docs.couchbase.com/styleguide/word-list.html#n)

```suggestion
{name-sdk} 1.1 adds support for JWT and client certificate authentication, as well as a new Server Async poll-based API that uses request handles to fetch results.
Introduced in the self-managed Enterprise Analytics Server 2.2, this API eliminates the need for long-running server connections.

The examples in this first section of the page are for the standard API -- async on the client side -- working with Enterprise Analytics 2.0 and later (with Server Async examples following in the <<#server-asynchronous-api,Server Async section>>).
You can still use this API with Enterprise Analytics 2.2 and later, in addition to the new API.
```

----

=== Server Asynchronous API

> In the 2.2 release, the Enterprise Analytics Server introduces an asynchronous request API.

Reviewer: As above. I'm not going to flag all usage of version numbers, just a reminder to make them consistent

Also, use the marketing name for the Server Async feature, when decided.

```suggestion
Enterprise Analytics 2.2 introduces an asynchronous request API.
```

----

=== Server Asynchronous API

In the 2.2 release, the Enterprise Analytics Server introduces an asynchronous request API.
The SDK sends a request, polls for results, and then fetches once the result is available.
The SDK supports each stage of this information flow:

> `cluster.start_query()` → `QueryHandle` → `QueryStatus` → `QueryResultsHandle`

Reviewer 2: ```suggestion
`cluster.StartQueryAsync()` → `QueryHandle` → `QueryStatus` → `QueryResultsHandle`
```
